lib/Support/Check.cpp

```diff
@@ -62,6 +78,23 @@ static bool testBufferFuzzy(offloadtest::Buffer *B1, offloadtest::Buffer *B2,
     }
     return true;
   }
+  case offloadtest::DataFormat::Float16: {
+    if (B1->Size != B2->Size)
+      return false;
+    llvm::MutableArrayRef<uint16_t> Arr1(
+        reinterpret_cast<uint16_t *>(B1->Data.get()),
+        B1->Size / sizeof(uint16_t));
+    assert(B2->Format == offloadtest::DataFormat::Float16 &&
+           "Buffer types must be the same");
+    llvm::MutableArrayRef<uint16_t> Arr2(
+        reinterpret_cast<uint16_t *>(B2->Data.get()),
+        B2->Size / sizeof(uint16_t));
+    for (unsigned I = 0; I < Arr1.size(); ++I) {
```

Comment:
It really doesn't matter much here, but it's a good habit to get in to avoiding calling a function to recalculate the end of a loop on every iteration.
```suggestion
    for (unsigned I = 0, E = Arr1.size(); I < E; ++I) {
```

See https://llvm.org/docs/CodingStandards.html#don-t-evaluate-end-every-time-through-a-loop for a longer rationale for this.